pandas/tests/series/methods/test_replace.py

```diff
@@ -675,3 +675,26 @@ def test_replace_value_none_dtype_numeric(self, val):
         result = ser.replace(val, None)
         expected = pd.Series([1, None], dtype=object)
         tm.assert_series_equal(result, expected)
+
+    def test_replace_change_dtype_series(self):
+        # GH#25797
+        df = pd.DataFrame.from_dict({"Test": ["0.5", True, "0.6"]})
+        print(df["Test"].dtype)
```

Comment:
Some pointers: Please remove the print statements.

Generally, you can remove your first assert statements per block, but we want to compare the whole df with the second assert statement, e.g. define your expected object and then call assert_frame_equal

```
expected = ...
tm.assert_frame_equal(df, expected)
```
 
if expected is the same as your initial df, you can define expected through

```
expected = df.copy()
```

in the beginning

Reply:
Thanks! Made the changes. 

Correct me if I am wrong, but here the assert needs to check if the series dtype remains 'object' and not if the dataframes are equal. I've made changes accordingly.

Reply:
tm.assert_frame_equal raises if your dtypes are different. So this is covered as well. In general, we want to check that the DataFrame is as expected (values, columns, index, dtype, ...). That is why we are using assert_frame_equal

The functions could modify the values unexpectedly and the test would not fail, if you only check the dtypes

Reply:
Ah okay! Thanks for explaining :)